sdks/python/apache_beam/runners/portability/fn_api_runner.py

```diff
@@ -277,6 +277,159 @@ def deduplicate_read(self):
 
     safe_coders = {}
 
+    def lift_combiners(stages):
+      """Expands CombinePerKey into pre- and post-grouping stages.
+
+      ... -> CombinePerKey -> ...
+
+      becomes
+
+      ... -> PreCombine -> GBK -> MergeAccumulators -> ExtractOutput -> ...
+      """
+      def new_id(existing_ids, prefix=''):
```

Comment:
<!--new_thread; commit:15c1178c61218f4f56ad3d74a1d52f3f1dc6e727; resolved:0-->
Looks like similar to unique_name at the end of the file. Shall we dedup them or if not, move helpers closer together?

Reply:
Good call. Done.